EngineLayer/Psm.cs

```diff
@@ -273,6 +273,13 @@ public void AddOrReplace(CompactPeptideBase compactPeptide, double score, int no
             }
         }
 
+        public void ReplaceAndAdd(List<Tuple<CompactPeptideBase, int>> cps)
+        {
+            compactPeptides = new Dictionary<CompactPeptideBase, Tuple<int, HashSet<PeptideWithSetModifications>>>();
+            foreach (Tuple<CompactPeptideBase, int> cp in cps)
+                compactPeptides[cp.Item1] = new Tuple<int, HashSet<PeptideWithSetModifications>>(cp.Item2, null);
+        }
```

Comment:
Replacement should only be done if found better scoring peptides. 

This method violates this assumption, allowing replacing the peptides arbitrarily. 

Can you refactor your code so that this method is not necessary?